statsmodels/gam/smooth_basis.py

```diff
@@ -792,6 +792,8 @@ def transform(self, x_new):
         basis : ndarray
             design matrix for the spline basis for given ``x_new``.
         """
+        if x_new.ndim == 1:
+            x_new = x_new.reshape(-1, 1)
```

Comment:
This causes another bug if 1-d is a row.
We need conditioning similar to the Results predict method

Reply:
this should work to distinguish row and column 1-dim:
`if x_new.ndim == 1 and self.k_variables == 1:`

self.k_variables is defined in `__init__`

Reply:
Gosh, that was a good spot! Fixed.